woocommerce/woocommerce
# Add Campaigns card in Multichannel Marketing page

### All Submissions:


### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/34905.

In this PR, we add a new Campaigns card into the Multichannel Marketing page. This builds on top of PR https://github.com/woocommerce/woocommerce/pull/36541, and uses the campaigns API introduced in PR https://github.com/woocommerce/woocommerce/pull/36222.

Campaigns card in loading state:

![image](https://user-images.githubusercontent.com/417342/221236817-69da0e27-3c71-4276-9968-25f4036806d1.png)

Campaigns card in error state:

![image](https://user-images.githubusercontent.com/417342/216299053-97c72615-1313-474c-9ebb-9a28f02edaea.png)

Campaigns card in empty data state (no campaigns):

![image](https://user-images.githubusercontent.com/417342/216299227-0392da4d-dfec-4c38-826c-5457c01f7de9.png)

Campaigns card with campaign data and pagination:

![image](https://user-images.githubusercontent.com/417342/216297756-e8d802e7-03cf-4370-ba94-d464fe06c53e.png)

Demo video showing pagination transition:

https://user-images.githubusercontent.com/417342/221238225-181b8370-d5fd-4bc8-a967-5d42c3280f1e.mp4

Demo video showing the page transition when we click on the campaign link:

https://user-images.githubusercontent.com/417342/216300467-bc5faab3-8a0e-4bbc-a29c-6c5334440df1.mp4

### How to test the changes in this Pull Request:

Pre-condition: 

- Enable WooCommerce Multichannel Marketing experience in WooCommerce Settings > Advanced > Features (`/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`).
- Use Google Listings and Ads plugin from this PR https://github.com/woocommerce/google-listings-and-ads/pull/1730.
- Complete the setup onboarding in Google Listings and Ads and create a few ads campaigns. Alternatively, you can also follow the test instructions in PR https://github.com/woocommerce/woocommerce/pull/36222 so that the campaigns API returns fake dummy campaigns data.

Test instructions:

1. Go to Marketing page at `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`.
2. You should see the Campaigns card at the top of the page. The Campaigns card should display different states like the screenshots above.
    - You can modify `useCampaigns` hook to return the desired states for testing, e.g. `loading: true`, `data: undefined`, `data: []` etc.
3. Click on the next and previous pagination buttons. It should work as expected. See the demo video above for how it should work.
4. Click on the campaign link. It should bring you to the campaign page quickly via client-side navigation.

### Other information:


### FOR PR REVIEWER ONLY:
